packages/sns-ingest/index.js

```diff
@@ -0,0 +1,118 @@
+'use strict'
+
+const local = require('kes/src/local')
+const satlib = require('@sat-utils/api-lib')
+const request = require('request')
+
+function downloadJSON(url) {
+  return new Promise((resolve, reject) => {
+    request(url, (error, response, body) => {
+      if (error) reject(error)
+      if (response.statusCode !== 200) {
+        reject(`Invalid status code <${response.statusCode}>`)
+      }
+      resolve(body)
+    })
+  })
+}
+
+function handler(event, _context, _cb) {
+  //console.log(JSON.stringify(event))
+
+  // Build array with all STAC items
+  const importedStacItems = []
+  event.Records.forEach((record) => {
+    // load stac item
+    const body = JSON.parse(record.body)
+    const importedStacItem = JSON.parse(body.Message)
+    console.log(importedStacItem)
+    importedStacItems.push(importedStacItem)
+    //console.log(importedStacItem.geometry.coordinates[0][0][0][0])
+    //console.log(importedStacItem.geometry.coordinates[0][0][0][1])
+    //console.log(importedStacItem.geometry.coordinates[0][0][1][0])
+    //console.log(importedStacItem.geometry.coordinates[0][0][1][1])
+  })
+
+  satlib.es.client().then((client) => {
+    console.log('Connected to ES')
+
+    // Create a 'collection' index in ES
+    satlib.es.putMapping(client, 'collections').catch((_err) => {})
+
+    // traverse list and add collection if necessary
+    // satlib.es.client is an async function
+    importedStacItems.forEach((importedStacItem) => {
+      // Check if we need to import collection
+      console.log('Checking ', importedStacItem.properties['c:id'])
+      client.exists({
+        index: 'collections',
+        type: 'doc',
+        id: importedStacItem.properties['c:id']
+      }).then((collectionExists) => {
+        if (!collectionExists) {
+          // Read collection information and insert into ES
+          // Load collection from stac item
+          downloadJSON(importedStacItem.links.collection.href).then((body) => {
+            const importedCollection = JSON.parse(body)
+            console.log(importedCollection)
+            console.log(importedCollection.properties['c:id'])
+
+            // @todo repeating c:id parameter at first level, check 'c:id' passed
+            // to saveRecords below
+            importedCollection['c:id'] = importedCollection.properties['c:id']
+
+            // @todo check, had to remove index= named parameter
+            satlib.es.saveRecords(client, [importedCollection], 'collections', 'c:id',
+              (err, _updated, _errors) => {
+                if (err) console.log('Error: ', err)
+              })
+          })
```

Comment:
This is one of the 'to do ' items, I'm  having to create a 'c:id' field at the document's root level in order to interface with satlib.es.saveRecords().